src/panels/lovelace/cards/hui-glance-card.ts

```diff
@@ -75,34 +82,53 @@ export class HuiGlanceCard extends hassLocalizeLitMixin(LitElement)
     const columns = config.columns || Math.min(config.entities.length, 5);
     this.style.setProperty("--glance-column-width", `${100 / columns}%`);
 
-    this.configEntities = entities;
+    this._configEntities = entities;
 
     if (this.hass) {
       this.requestUpdate();
     }
   }
 
-  protected render() {
-    if (!this.config || !this.hass) {
-      return html``;
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (changedProps.get("hass") && this._configEntities) {
+      for (const entity of this._configEntities) {
+        if (
+          (changedProps.get("hass") as any).states[entity.entity] !==
+          this.hass!.states[entity.entity]
+        ) {
+          return true;
+        }
+      }
+      return false;
     }
-    const { title } = this.config;
+    return (changedProps as unknown) as boolean;
+  }
 
-    applyThemesOnElement(this, this.hass!.themes, this.config.theme);
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+    const { title } = this._config;
 
     return html`
       ${this.renderStyle()}
       <ha-card .header="${title}">
         <div class="entities ${classMap({ "no-header": !title })}">
-          ${this.configEntities!.map((entityConf) =>
+          ${this._configEntities!.map((entityConf) =>
             this.renderEntity(entityConf)
           )}
         </div>
       </ha-card>
     `;
   }
 
-  private renderStyle() {
+  protected updated(_changedProperties: PropertyValues): void {
+    if (this.hass && this._config) {
```

Comment:
We should only apply theme on first render or when `this._config.theme` has changed or if `hass.themes` has changed.

Reply:
So using `firstUpdated` but for theme changes should we use updated and check the changed props there?

Reply:
On first update, both `firstUpdated` and `updated` are called. I think that instead we should just inside updated check for it:

```
if (!changedProps.has('hass') || changedProps.get('hass').themes !== this.hass.themes)
```

Reply:
Need to use `get('hass')` instead of `has` because it always has hass but hass is set to undefined (I guess not set but the lack of being set yet)

Reply:
that is odd. It is only part of changedProps if it was set to something by the parent.